### PR TITLE
New Spraypaint Bugfixes

### PR DESCRIPTION
**Pull request by PYROTEKNIK**
Should handle missing assets a little better now. The gap needed between sprays is now based on size, but the decal assets must be installed serverside for it to work. I think this is something the decal system needs in order to work properly anyway.

### CODE REVIEW

File: lua/weapons/weapon_stencilpaint.lua [hunk 1]
```diff
@@ -17,102 +17,37 @@ end
...
     list.Set("SprayPaintStencils", 41, "Noughtsncrosses")
     list.Set("SprayPaintStencils", 42, "Nought")
     list.Set("SprayPaintStencils", 43, "Cross")
     list.Set("SprayPaintStencils", 44, "Eye")
     list.Set("SprayPaintStencils", 45, "Smile")
-end
+    
+end 
+
+hook.Add("InitPostEntity", "StencilPaint_RegisterDecals", CreateDecals())
```
Comment (swampservers) [inline lua/weapons/weapon_stencilpaint.lua]: I think you mean: `hook.Add("InitPostEntity", "StencilPaint_RegisterDecals", CreateDecals)`

If you make a function that is only for one hook, please just make the function anonymous inside the hook:

```
hook.Add("InitPostEntity", "StencilPaint_RegisterDecals", function()
...
end)
```

**Comment by swampservers** [comment]
you put the .tga and .pdn (whatever that is...) in again

**Comment by swampservers** [comment]
you can edit the .gitignore if you want to keep them in your working dir (probably dont' commit .gitignore change though)

**Comment by PYROTEKNIK** [comment]
can you make gitignore ignore itself?

also pdn is a paint.net file, it's just the split layers

**Pull request merged**